### ISSUE TITLE
Cannot install firebase-admin v10

### ISSUE THREAD

**Issue by Smrtnyk** [firebase/firebase-functions#996]
### [REQUIRED] Step 2: Describe your environment

  * Operating System version: Windows 11
  * Firebase SDK version: firebase-admin 10
  * Firebase Product: firebase-admin 10
  * Node.js version: 16.8.0
  * NPM version: 7.21.0

#### Steps to reproduce:

After increasing firebase-admin to v 10 from 9.12.0 I get errors about peer dependencies that are not met for firebase-functions.
Seems like firebase-functions require firebase-admin v9.

#### Relevant Code:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: firebase-functions@3.15.7
npm ERR! Found: firebase-admin@10.0.0
npm ERR! node_modules/firebase-admin
npm ERR!   firebase-admin@"10.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer firebase-admin@"^8.0.0 || ^9.0.0" from firebase-functions@3.15.7
npm ERR! node_modules/firebase-functions
npm ERR!   firebase-functions@"3.15.7" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: firebase-admin@9.12.0
npm ERR! node_modules/firebase-admin
npm ERR!   peer firebase-admin@"^8.0.0 || ^9.0.0" from firebase-functions@3.15.7
npm ERR!   node_modules/firebase-functions
npm ERR!     firebase-functions@"3.15.7" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

```
my package.json
```
"dependencies": {
        "diff-arrays-of-objects": "1.1.9",
        "firebase-admin": "10.0.0",
        "firebase-functions": "3.15.7",
        "web-push": "3.4.5"
    },
    "devDependencies": {
        "@types/faker": "5.5.8",
        "@types/web-push": "3.3.2",
        "@typescript-eslint/eslint-plugin": "5.0.0",
        "@typescript-eslint/parser": "5.0.0",
        "eslint": "8.0.0",
        "eslint-plugin-import": "2.25.2",
        "faker": "5.5.3",
        "typescript": "4.4.4"
    },
 ```

**Comment by Smrtnyk** [comment]
> The Functions SDK (firebase-functions) requires firebase-admin v8 or v9:
> 
> ```
> peer firebase-admin@"^8.0.0 || ^9.0.0"
> ```
> 
> So v10 won't really work with firebase-functions until they update their peering requirement.
> 
> On top of that it seems you also have firebase-admin 9.12.0 currently installed in your work environment:
> 
> ```
> Conflicting peer dependency: firebase-admin@9.12.0
> ```

That is because I didn't delete node_modules, but just bumped the version in the package.json and hit npm install
when I delete node_modules and package.lock still same issue ocurrs.
Since it is something for firebase-functions to do, feel free to close this issue then

**Comment by inlined** [comment]
Will be fixed in #998 which should be released early next week.
